refactor(authenticator)!: use Amplify JS auto sign in (#2642)
aws-amplify/amplify-ui: docs/src/pages/[platform]/getting-started/migration/migration.angular.mdx

import { Tabs, TabItem } from '@aws-amplify/ui-react';
import { Alert } from '@aws-amplify/ui-react';

## 2.x to 3.x
### Installation

> Install the 3.x version of the `@aws-amplify/ui-angular` library.

Reviewer: ```suggestion
Install the 3.x version of the `@aws-amplify/ui-angular` and 5.x version of the `aws-amplify` libraries.
```

Author: Done

<Tabs>
<TabItem title="npm">

```shell
npm install aws-amplify@5.x @aws-amplify/ui-angular@3.x
```

</TabItem>
<TabItem title="yarn">
```shell
yarn add aws-amplify@5.x @aws-amplify/ui-angular@3.x
```

</TabItem>
</Tabs>

### Update and usage

The following breaking changes were introduced in `@aws-amplify/ui-angular` major version `3.0`:

#### 1. Automatic signin on signup logic moved to `aws-amplify`. 

> If you are overriding `Auth.signUp`, update the override function call to include the `autoSignIn` option set to enabled. If this change is not made, users will not be automatically signed in on signup.

Reviewer 2: nit: 
```suggestion
If you are overriding `Auth.signUp`, update the override function call to include the `autoSignIn` option set to `enabled`. If this change is not made, users will not be automatically signed in on signup.
```

Author: ok!

import { Tabs, TabItem } from '@aws-amplify/ui-react';
import { Alert } from '@aws-amplify/ui-react';

## 2.x to 3.x
### Installation
Install the 3.x version of the `@aws-amplify/ui-angular` library.

<Tabs>
<TabItem title="npm">

```shell
npm install aws-amplify @aws-amplify/ui-angular@3.x
```

</TabItem>
<TabItem title="yarn">
```shell
yarn add aws-amplify @aws-amplify/ui-angular@3.x
```

</TabItem>
</Tabs>

### Update and usage

The following breaking changes were introduced in `@aws-amplify/ui-angular` major version `3.0`:

> #### 1. Update the handleSignUp function override call

Reviewer 3: It feels like these two steps should be under a heading:

```
#### Automatic signin on signup logic moved to `aws-amplify`. 

Make the following changes to upgrade to 3.0:

##### 1. Update the handleSignUp function override call
...
```

Author: 👍

> Update the override function call to include the `autSignIn` set to enabled. This must be included if you are overriding `Auth.signUp`.

Reviewer 3: See comment above in vue migration

Author: 👍

## 2.x to 3.x
### Installation
Install the 3.x version of the `@aws-amplify/ui-angular` library.

<Tabs>
<TabItem title="npm">

```shell
npm install aws-amplify @aws-amplify/ui-angular@3.x
```

</TabItem>
<TabItem title="yarn">
```shell
yarn add aws-amplify @aws-amplify/ui-angular@3.x
```

</TabItem>
</Tabs>

### Update and usage

The following breaking changes were introduced in `@aws-amplify/ui-angular` major version `3.0`:

#### Automatic signin on signup logic moved to `aws-amplify`. 

##### 1. Update the handleSignUp function override call

If you are overriding `Auth.signUp`, update the override function call to include the `autoSignIn` option set to enabled. If this change is not made, users will not be automatically signed in on signup.

```diff
 async handleSignUp(formData) {
  let { username, password, attributes } = formData;
  // custom username
  username = username.toLowerCase();
  attributes.email = attributes.email.toLowerCase();
  return Auth.signUp({
    username,
    password,
    attributes,
+   autoSignIn: {
+     enabled: true
+   }
  });
}

```
##### 2. Update to the latest version of the `aws-amplify` library.

> Update to the latest version of the `aws-amplify` library. You will need at least version `^4.3.29`. 

Reviewer 3: Update this to 5.0, since that's the version which will be required by our major version bump

Author: 👍

## Breaking changes

The 2.x version of the `Authenticator` component has several differences from earlier versions. Here are a few of the major changes that you'll need to look out for.

### Slots

All the slot locations have changed with the 2.x version of the `Authenticator`. To get a
sense of the changes please check out the [Headers and Footers](../connected-components/authenticator/customization#headers--footers) section.

### Form Fields

> The 2.x version of the `Authenticator` has a different format for the `formFields` prop. It also no long accepts

Reviewer 2: ```suggestion
The 2.x version of the `Authenticator` has a different format for the `formFields` prop. It also no longer accepts
```

Author: Fixed to longer

> `inputProps` nor `hint`. Instead it's recommended that you use the [Headers and Footers Slots](../connected-components/authenticator/customization#headers--footers) or use the

Reviewer 2: Nit:

```suggestion
`inputProps` nor `hint`. Instead, it's recommended that you use the [Headers and Footers Slots](../connected-components/authenticator/customization#headers--footers) or use the
```

Author: added comma
